### PR TITLE
Whitelising tests

### PR DESCRIPTION
**Pull request by krzkaczor**

### CODE REVIEW

File: test/helpers/increaseTime.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default function increaseTime(duration) {
 }
 
 // sets time to given timestamp based on current block time
-export function setTimeTo(timestamp) {
+export async function setTimeTo(timestamp) {
```
Comment (Recmo) [inline test/helpers/increaseTime.js]: This function does not use `await`, so doesn't need `async`. This whole file could use a good rewrite, starting with a `toPromise` wrapper function for `web3` functions.

Comment (Recmo) [inline test/helpers/increaseTime.js]: Or maybe using the synchronous `send` functions like in `latestTimestamp()`.

File: test/helpers/latestTime.js [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+export const DAY = 24 * 60 * 60;
+export const MONTH = HOUR * 24 * 31;
+
+// useful for spawning time sensitive contracts
+export function closeFutureDate() {
+  return latestTimestamp() + MONTH + MONTH;
+}
+
+// useful for spawning time sensitive contracts
+export function furterFutureDate() {
```
Comment (Recmo) [inline test/helpers/latestTime.js]: Typo `furtHer`

File: test/helpers/latestTime.js [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+  return latestTimestamp() + MONTH;
```
Comment (Recmo) [inline test/helpers/latestTime.js]: Doesn't this make the further date closer than the closer date?

File: test/helpers/verification.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+let curve;
+
+export async function curveInEur(moneyInEur) {
+  if (!curve) {
+    curve = await deployCurve();
+  }
+
+  return curve.curve(moneyInEur);
```
Comment (Recmo) [inline test/helpers/verification.js]: The function argument is in Euro Units of Least Precision (EuroULP), which is 10^-18 of a euro.

File: test/helpers/verification.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export async function curveInEther(money, eurEtherRatio) {
+  if (!curve) {
+    curve = await deployCurve();
+  }
+
+  const moneyInEur = ethToEur(money, eurEtherRatio);
+
+  return curve.curve(moneyInEur);
```
Comment (Recmo) [inline test/helpers/verification.js]: same here

File: test/helpers/verification.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export function ethToEur(ether, eurEtherRatio = etherToWei(218.1192809)) {
```
Comment (Recmo) [inline test/helpers/verification.js]: Make it 300. At least :p

File: test/helpers/increaseTime.js (hunk 1 above)
Comment (krzkaczor) [inline test/helpers/increaseTime.js]: i added async there to make it clear that it's async ie. it returns a promise. I was a source of a bug in my previous tests

File: test/helpers/chaiWeb3.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    // I would love to hear ideas for better implementation
+
+    const web3ArrayAccessor = this._obj;
+
+    try {
+      await web3ArrayAccessor(size - 1);
+    } catch (e) {
+      this.assert(
+        false,
+        `expected web3 array to be size of ${size} but it looks like it's smaller` // i think it's impossible to get an array name in this point
```
Comment (Recmo) [inline test/helpers/chaiWeb3.js]: > impossible

```
    const req = web3ArrayAccessor.request(0);
    const bytes4 = req.params[0].data.slice(10);
```

The bytes4 contains a hash of the array name, you can now iterate through all contracts and find the matching one :D

File: test/helpers/increaseTime.js (hunk 1 above)
Comment (Recmo) [inline test/helpers/increaseTime.js]: Yeah, my main problem is the inconsistency with `increaseTime` before it. That one should also have an `async` flag then.

Or, alternatively, replace `sendAsync` with `send` and make both function synchronous. This is how it is done elswhere (specifically in latestTimestamp). Again, for consistency we should do the same everywhere.

Comment (Recmo) [inline test/helpers/increaseTime.js]: I think I prefer the synchronous approach, has the least opportunity to screw up.

Comment (krzkaczor) [inline test/helpers/increaseTime.js]: Yeah, let's agree on adding async keyword to `increaseTime`. 

Redoing it for sync calls is also a great idea but i don't want to do any more refactoring on this branch. Let's discuss improving things like this tmrw.

File: test/helpers/chaiWeb3.js (hunk 7 above)
Comment (krzkaczor) [inline test/helpers/chaiWeb3.js]: 👍 i hope one day i will find courage to implement it xD

**Pull request merged**